var/spack/repos/builtin/packages/mptensor/package.py

```diff
@@ -70,31 +70,30 @@ def setup_build_tests(self):
         install test subdirectory for use during `spack test run`."""
         self.cache_extra_test_sources(".")
 
-    def test(self):
+    def test_mpi(self):
+        """test with +mpi"""
         if "+mpi" not in self.spec:
-            print("Test of mptensor only runs with +mpi option.")
-        else:
-            with working_dir(join_path(self.install_test_root, "tests"), create=False):
-                make("clean")
-                makefile = FileFilter("Makefile")
-                makefile.filter("g++", "{0}".format(spack_cxx), string=True)
-
-            with working_dir(join_path(self.install_test_root), create=False):
-                makefile = FileFilter("Makefile.option")
-                makefile.filter("CXX =.*", "CXX ={0}".format(self.spec["mpi"].mpicxx))
-                makefile.filter("CXXFLAGS =.*", "CXXFLAGS ={0}".format(self.compiler.cxx11_flag))
-
-            math_libs = (
-                self.spec["scalapack"].libs + self.spec["lapack"].libs + self.spec["blas"].libs
-            )
+            raise SkipTest("Package must be installed with +mpi")
+
+        with working_dir(join_path(self.install_test_root, "tests"), create=False):
+            make("clean")
+            makefile = FileFilter("Makefile")
+            makefile.filter("g++", "{0}".format(spack_cxx), string=True)
+
+        with working_dir(join_path(self.install_test_root), create=False):
+            makefile = FileFilter("Makefile.option")
+            makefile.filter("CXX =.*", "CXX ={0}".format(self.spec["mpi"].mpicxx))
+            makefile.filter("CXXFLAGS =.*", "CXXFLAGS ={0}".format(self.compiler.cxx11_flag))
```

Comment:
This cleanup should be done *once* in `setup_build_tests`.  See #38633 for a draft that makes this change.